src/main/java/org/codehaus/plexus/archiver/jar/JarArchiver.java

```diff
@@ -287,6 +298,34 @@ public void setFilesetmanifest( FilesetManifestConfig config )
     }
 
     /**
+     * Sets the configuration for the module inside modular JAR file.
+     * The module descriptor will be extended with the
+     * configured metadata such as module version and main class.
+     * <p/>
+     * If set to {@code null} the module descriptor (if present)
+     * will be archived as is.
+     *
+     * @param moduleConfiguration The module configuration
+     *
+     * @since 3.6
+     */
+    public void setModuleConfiguration( ModuleConfiguration moduleConfiguration )
```

Comment:
What about WAR and EAR files? What do you think - should they throw exception if they are called with non-null values. Maybe I'm wrong but I don't think there is such thing as a modular WAR file.

Also what about if you set `ModuleConfiguration` for non-modular JAR file. Maybe we should throw exception as it may indicate configuration error.